circle.yml

```diff
@@ -1,9 +1,14 @@
+machine:
+  environment:
+    PATH: ~/perl5/bin:$PATH
 dependencies:
-  post:
-    - curl -L https://cpanmin.us | perl - --sudo App::cpanminus
-    - cpanm -nS local::lib
-    - cpanm -nS TAP::Parser
-    - cpanm -nL locallib --installdeps .
+  cache_directories:
+    - ~/perl5
+  pre:
+    - if ! hash perl 2>/dev/null; then sudo apt-get update -qq; sudo apt-get install perl; fi
+    - if ! hash cpanm 2>/dev/null; then curl -L https://cpanmin.us | perl - -l ~/perl5 App::cpanminus local::lib; fi
+    - if ! grep -q local::lib ~/.bashrc; then echo "eval $(perl -I ~/perl5/lib/perl5/ -Mlocal::lib)" >> ~/.bashrc; fi
+    - cpanm --quiet --notest --skip-satisfied --skip-installed --installdeps .
 test:
-  post:
-    - prove -lr -j 4 -Mlocal::lib=locallib t/*
\ No newline at end of file
+  override:
+    - prove --lib --recurse --jobs 4 t/*
```

Comment:
Favoring verbose flags to make the configuration more palatable to non-Perl developers.